src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -589,6 +640,58 @@ void FindEnzymesDialog::accept() {
     QDialog::accept();
 }
 
+void FindEnzymesDialog::sl_handleSupplierSelectionChange(QStringList checkedSuppliers) {
+    const auto enzymes = EnzymesSelectorWidget::getLoadedEnzymes();
+    QList<SEnzymeData> visibleEnzymes;
+    for (const auto& enzyme : qAsConst(enzymes)) {
+        for (const auto& supplier : qAsConst(enzyme->suppliers)) {
+            CHECK_CONTINUE(checkedSuppliers.contains(supplier));
+
+            visibleEnzymes.append(enzyme);
+            break;
+
+        }
+    }
+    enzSel->setEnzymesList(visibleEnzymes);
+    static const QString notDefinedTr = EnzymesIO::tr(EnzymesIO::NOT_DEFINED_SIGN);
+    if (checkedSuppliers.contains(notDefinedTr)) {
+        checkedSuppliers.replace(checkedSuppliers.indexOf(notDefinedTr), EnzymesIO::NOT_DEFINED_SIGN);
+    }
+    auto value = checkedSuppliers.join(SUPPLIERS_LIST_SEPARATOR);
+    AppContext::getSettings()->setValue(EnzymeSettings::CHECKED_SUPPLIERS, value);
+}
+
+void FindEnzymesDialog::sl_updateSuppliers() {
+    const auto& loadedSuppliers = EnzymesSelectorWidget::getLoadedSuppliers();
+    cbSuppliers->clear();
+    cbSuppliers->addItems(loadedSuppliers);
+    QString selStr = AppContext::getSettings()->getValue(EnzymeSettings::CHECKED_SUPPLIERS).toString();
+    static const QString notDefinedTr = EnzymesIO::tr(EnzymesIO::NOT_DEFINED_SIGN);
+    cbSuppliers->setCheckedItems(selStr.isEmpty() ? loadedSuppliers : selStr.split(SUPPLIERS_LIST_SEPARATOR));
+    selStr = selStr.replace(EnzymesIO::NOT_DEFINED_SIGN, notDefinedTr);
```

Comment:
This code looks wrong (a typo?)
Here what you should do here:
1) Get list of the original suppliers from (loaded suppliers)
2) Replace the "NOT_DEFINED" with a translated value in the list. Set it to the UI.
3) Get CHECKED_SUPPLIERS from the settings: these are untranslated values. Split them into tokens. Replace the NOT_DEFINED token with a translated value
4) Call setCheckedItem().